props/saber_fett263_buttons.h

```diff
@@ -293,248 +659,3857 @@ class SaberFett263Buttons : public virtual PropBase {
+        if (millis() - push_begin_millis_ > saved_gesture_control.forcepushlen) {
+          // Checking for Clash at end of movement
+          clash_type_ = CLASH_CHECK;
+          push_begin_millis_ = millis();
+          clash_impact_millis_ = millis();
+        } 
+      } else {
+        push_begin_millis_ = millis();
+      }
+      
+    } else {
+      // EVENT_SWING - Swing On gesture control to allow fine tuning of speed needed to ignite
+      if (millis() - saber_off_time_ < MOTION_TIMEOUT) {
+        SaberBase::RequestMotion();
+        if (swinging_ && fusor.swing_speed() < 90) {
+          swinging_ = false;
+        }
+        if (!swinging_ && fusor.swing_speed() > saved_gesture_control.swingonspeed) {
+          swinging_ = true;
+          Event(BUTTON_NONE, EVENT_SWING);
+        }
+      }
+      // EVENT_THRUST
+      if (mss.y * mss.y + mss.z * mss.z < 16.0 &&
+          mss.x > 14  &&
+          fusor.swing_speed() < 150) {
+        if (millis() - thrust_begin_millis_ > 15) {
+          Event(BUTTON_NONE, EVENT_THRUST);
+          thrust_begin_millis_ = millis();
+        } 
+      } else {
+        thrust_begin_millis_ = millis();
+      }
+    }
+    if (menu_ || color_mode_ == CC_COLOR_LIST) {
+      DetectMenuTurn();
+    }
+    if (track_mode_ != PLAYBACK_OFF) {
+      TrackPlayer();
+    }
+  }
+
+  // Fett263 Track Player
+  enum TrackMode {
+    PLAYBACK_OFF,
+    PLAYBACK_LOOP,
+    PLAYBACK_ROTATE,
+    PLAYBACK_RANDOM,
+  };
+
+  void TrackPlayer() {
+    if (!track_player_) {
+#ifdef ENABLE_AUDIO
+      if (track_num_ == -1 && track_mode_ == PLAYBACK_LOOP) {
+        StartOrStopTrack();
+      } else {
+        if (track_num_ > num_tracks_) track_num_ = 0;
+        switch (track_mode_) {
+          case PLAYBACK_ROTATE:
+            track_num_ += 1;
+            break;
+          case PLAYBACK_RANDOM:
+            track_num_ = rand() % num_tracks_;
+            if (track_num_ < 0) track_num_ = num_tracks_ - 1;
+            break;
+          default:
+            break;
+        }
+        char playtrack[128];
+        RunCommandAndGetSingleLine("list_current_tracks", nullptr, track_num_, playtrack, sizeof(playtrack));
+        MountSDCard();
+        EnableAmplifier();
+        track_player_ = GetFreeWavPlayer();
+        if (track_player_) {
+          track_player_->Play(playtrack);
+        } else {
+          STDOUT.println("No available WAV players.");
+        }
+      }
+#else
+      STDOUT.println("Audio disabled.");
+#endif
+    }
+  }   
+   
+  void DetectMenuTurn() {
+     if (millis() - last_rotate_ > 1000) {
+      float a = fusor.angle2() - current_menu_angle_;
+      if (a > M_PI) a-=M_PI*2;
+      if (a < -M_PI) a+=M_PI*2;
+      change_ = a;
+      if (a > twist_menu_ * 2/3) {
+        current_menu_angle_ += twist_menu_;
+        if (current_menu_angle_ > M_PI) current_menu_angle_ -= M_PI * 2;
+        Event(BUTTON_NONE, EVENT_TWIST_RIGHT);
+        current_menu_angle_ = fusor.angle2();
```

Comment:
This is weird. Why do you reset the angle here?
Also, if you are going to reset the angle, what's the point of updating on lines 1423 and 1424?